### DISCUSSION TITLE
How to help Jakarta EE 11 to have JDK17 and JDK21 based implementation usable for validation of the TCK

### DISCUSSION THREAD

**Discussion by dmatej** [eclipse-ee4j/glassfish#24798] [Ideas]
In https://github.com/eclipse-ee4j/glassfish/discussions/24790 we voted that we really don't want to support JDK17 in GlassFish8. However Jakarta EE 11 spec lead companies voted that they want to support JDK17, however it is not mandatory for all implementations. That brings a problem how to validate the TCK as GlassFish was always used as the first passing implementation.

One option is to use GF8+Java21 and some other implementation with Java17.

Another option I am thinking about for few days is to go for Java21 and GlassFish8, and after it passes, we would create a special branch and distribution of GlassFish compatible with JDK17. It would not be so expensive as supporting both at once, GlassFish development would not be slowed down, and this GlassFish version would not be supported except the TCK. We would also expect that companies which voted for Java17 support in Jakarta EE 11 would contribute in such branch.

What do you think about it?

Relevant links:

- https://www.eclipse.org/lists/jakarta.ee-spec/msg03188.html
- https://jakartaee.github.io/platform/jakartaee11/JakartaEE11ReleasePlan
- https://www.eclipse.org/lists/jakarta.ee-spec.committee/msg03444.html

![image](https://github.com/eclipse-ee4j/glassfish/assets/302904/9a6339ec-055f-4828-b7ae-74d8de21fa08)

**Comment by dmatej** [comment]
Based on email discussions in relevant links it seems that GlassFish can safely continue with JDK21. At this time there is no need to support JDK17 too; TCK will be validated against JDK21. (probably).

**Comment by anbusampath** [comment]
> we would create a special branch and distribution of GlassFish compatible with JDK17.

> this GlassFish version would not be supported except the TCK.

This seems good option if no other implementation willing to support JDK 17 at Jakarata EE 11 release timeline.

**Comment by hs536** [comment]
I'm late in noticing this discussion, but I agree with making GF8+JDK21 the master and GF8+JDK17 the exceptional distribution.
